src/mon/AuthMonitor.cc

```diff
@@ -1682,14 +1673,8 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
           ds << entity_auth.key;
         }
       } else {
-	KeyRing kr;
-	kr.add(entity, entity_auth.key, entity_auth.pending_key);
-        if (f) {
-          kr.set_caps(entity, entity_auth.caps);
-          kr.encode_formatted("auth", f.get(), rdata);
-        } else {
-          kr.encode_plaintext(rdata);
-        }
+	_encode_key(entity, entity_auth, rdata, f.get(), true,
+		    &entity_auth.caps);
```

Comment:
1. All params to `_encode_key()` are specified – `pending_key=true`, `std::map<std::string, bufferlist>* caps` is NOT `nullptr`.
2. As `not pending_key` evaluates to `false`, the `kr.add(entity, eauth.key, eauth.pending_key)` path is taken.
3. The `caps` param is forwarded: `_encode_keyring(kr, entity, rdata, fmtr, caps)`.
4. As `caps cannot be `nullptr`, we may do `kr.set_caps(entity, *caps)` if `fmt` is not `nullptr`.

OK.